File: plugin/src/org/jetbrains/rust/highlight/RustHighlighterFactory.java
package org.jetbrains.rust.highlight;

import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory;
import org.jetbrains.annotations.NotNull;

public class RustHighlighterFactory extends SingleLazyInstanceSyntaxHighlighterFactory {

  @NotNull
  protected RustHighlighter createHighlighter() {
    return new RustHighlighter();
  }

}

File: plugin/src/org/jetbrains/rust/highlight/RustHighlighterFactory.kt
package org.jetbrains.rust.highlight

import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory

public class RustHighlighterFactory : SingleLazyInstanceSyntaxHighlighterFactory() {

    override fun createHighlighter(): RustHighlighter {
        return RustHighlighter()
    }

}